### ISSUE TITLE
aws_api_gateway_intergration_response : Method response not found

### ISSUE THREAD

**Issue by rjinski** [hashicorp/terraform#7486]
### Terraform Version

Terraform v0.6.16
### Affected Resource(s)
- aws_api_gateway_method
- aws_api_gateway_integration
### Terraform Configuration Files

https://gist.github.com/rjinski/b2324266e16f1b79fca25b3ba24e39ed
### Debug Output

Error applying plan:

1 error(s) occurred:
- aws_api_gateway_integration_response.demo: Error creating API Gateway Integration Response: NotFoundException: No integration defined for method
  status code: 404, request id: 28168313-428b-11e6-ada4-e1f298d71627

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
### Expected Behavior

All resources created successfully.
### Actual Behavior

When running `terraform apply` for the first time the above error is produced stating the the api gateway method resource could not be found however the resource on AWS has been created. Re-running `terraform apply` all resources are created.

**Comment by OJFord** [comment]
I had the same issue, add:
```hcl
depends_on [
    "aws_api_gateway_integration.your_integration"
]
```

It would be nice though if this could be automatically taken care of, e.g. by collecting all resources for a method, and doing the integration response last.

**Comment by Ninir** [comment]
Hi @rjinski 

Added a warning on this in https://github.com/hashicorp/terraform/pull/8049.
While the documentation is updated, the code can also be improved.

**Comment by stack72** [comment]
Can this be closed because of #8049?

**Comment by OJFord** [comment]
@stack72 I think *ideally* the behaviour could be improved, such that the docs-only change in #8049 could be reverted.
